core/tests/puppeteer-acceptance-tests/spec/curriculum-admin-tests/create-and-publish-topics-and-stories.spec.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright 2024 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Acceptance Test for topic management by curriculum admin
+ */
+
+import {UserFactory} from '../../puppeteer-testing-utilities/user-factory';
+import testConstants from '../../puppeteer-testing-utilities/test-constants';
+import {CurriculumAdmin} from '../../user-utilities/curriculum-admin-utils';
+
+const DEFAULT_SPEC_TIMEOUT = testConstants.DEFAULT_SPEC_TIMEOUT;
+const ROLES = testConstants.Roles;
+
+describe('Curriculum Admin', function () {
+  let curriculumAdmin: CurriculumAdmin;
+  let explorationUrl: string | null;
+  let explorationId: string;
+
+  let exploration = {
```

Comment:
In the top level test file, you shouldn't have all this data and the getExplorationIdFromUrl. The top-level test files should read like a user journey spec and not have technical detail. 

Please move getExplorationIdFromUrl to a suitable place (e.g. a general-purpose url parser utility service or something that the base user can do). For things like createTopic etc. you should pass arguments explicitly (e.g. `createTopic('Test Topic 1', 'test-topic-one')`) and autogenerate the web_fragment and description from the given arguments if they're not that important for the purposes of the test (and if they're important, pass them explicitly as additional arguments).